packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/common/TextNormalizer.ts

```diff
@@ -0,0 +1,266 @@
+import { TextSpan } from '../../types';
+import { END, spanLen, START } from './textSpanUtils';
+
+type SpanMapping = { rawSpan: TextSpan; normalizedSpan: TextSpan };
+
+const SPACES = {
+  normal: () => ' ',
+  regexString: '\\s+'
+};
+
+const DOUBLE_QUOTE = {
+  normal: () => '"',
+  regexString: `[${[
+    '«', // U+00AB
+    '»', // U+00BB
+    '“', // U+201C
+    '”', // U+201D
+    '„', // U+201E
+    '‟', // U+201F
+    '❝', // U+275D
+    '❞', // U+275E
+    '⹂', // U+2E42
+    '〝', // U+301D
+    '〞', // U+301E
+    '〟', // U+301F
+    '＂' // U+FF02
+  ].join('')}]`
+};
+
+const QUOTE = {
+  normal: () => "'",
+  regexString: `[${[
+    '‹', // U+2039
+    '›', // U+203A
+    '’', // U+2019
+    '❮', // U+276E
+    '❯', // U+276F
+    '‘', // U+2018
+    '‚', // U+201A
+    '‛', // U+201B
+    '❛', // U+275B
+    '❜', // U+275C
+    '❟' // U+275F
+  ].join('')}]`
+};
+
+const SURROGATE_PAIR = {
+  normal: (_: string) => '_',
+  regexString: '[\uD800-\uDBFF][\uDC00-\uDFFF]'
+};
+
+// remove "Combining Diacritical Marks" from the string
+// NOTE: we may have to do this after conversion again
+// str.normalize("NFD").replace(/[\u0300-\u036f]/g, "")
+const DIACRITICAL_MARK = {
+  normal: () => '',
+  regexString: '[\u0300-\u036f]'
+};
+const DIACRITICAL_MARK_REGEX = new RegExp(DIACRITICAL_MARK.regexString, 'g');
+
+function normalizeDiacriticalMarks(text: string, keepLength = false) {
+  const r = text
+    .normalize('NFD')
+    .replace(DIACRITICAL_MARK_REGEX, DIACRITICAL_MARK.normal)
+    .normalize('NFC');
+  if (keepLength && r.length !== text.length) {
+    //
+    // String.normalize may change length of a string. `keepLength` flag keeps string
+    // length after conversion by padding or truncating a string.
```

Comment:
It rarely happen. The last step of getting the result string is `normalize('NFC')`. It basically shorten the original text. I added `substring` just in case.